ci/lib/setup.sh

```diff
@@ -0,0 +1,81 @@
+# configure cwd, vars and logging
+_setup_env() {
+  # -ET: propagate DEBUG/RETURN/ERR traps to functions and subshells
+  set -ET
+  # exit on unhandled error
+  set -o errexit
+  # exit on unset variable
+  set -o nounset
+  # pipefail: any failure in a pipe causes the pipe to fail
+  set -o pipefail
+
+  if [[ -n "${SCRIPT_DEBUG:-}" ]]; then
+    set -o xtrace
+    # http://www.skybert.net/bash/debugging-bash-scripts-on-the-command-line/
+    export PS4='# ${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:-}() - [${SHLVL},${BASH_SUBSHELL},$?] '
+  fi
+  trap _err_trap ERR
+  # shellcheck disable=SC2034
+  # START_DIR is used elsewhere.
```

Comment:
where/how is this used?

Reply:
Oops, that's in fact _not_ used here, but is over inside a different codebase. Will remove.

Reply:
Removed it.